pkg/deployments/tasks/20220725-protocol-fee-percentages-provider/test/task.fork.ts

```diff
@@ -0,0 +1,92 @@
+import hre from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+
+import { BigNumber, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import { describeForkTest } from '../../../src/forkTests';
+import Task, { TaskMode } from '../../../src/task';
+import { getForkedNetwork } from '../../../src/test';
+import { getSigner, impersonate } from '../../../src/signers';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+
+describeForkTest('ProtocolFeePercentagesProvider', 'mainnet', 15130000, function () {
+  let admin: SignerWithAddress;
+  let protocolFeePercentagesProvider: Contract;
+  let vault: Contract, authorizer: Contract, feesCollector: Contract;
+
+  let task: Task;
+
+  enum FeeType {
+    Swap = 0,
+    FlashLoan = 1,
+    Yield = 2,
+    AUM = 3,
+  }
+
+  before('run task', async () => {
+    task = new Task('20220725-protocol-fee-percentages-provider', TaskMode.TEST, getForkedNetwork(hre));
+    await task.run({ force: true });
+    protocolFeePercentagesProvider = await task.deployedInstance('ProtocolFeePercentagesProvider');
+  });
+
+  before('setup accounts', async () => {
+    admin = await getSigner(0);
+  });
+
+  before('setup contracts', async () => {
+    const vaultTask = new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    vault = await vaultTask.deployedInstance('Vault');
+    authorizer = await vaultTask.instanceAt('Authorizer', await vault.getAuthorizer());
+    feesCollector = await vaultTask.instanceAt('ProtocolFeesCollector', await vault.getProtocolFeesCollector());
+  });
+
+  before('setup admin', async () => {
+    const DEFAULT_ADMIN_ROLE = await authorizer.DEFAULT_ADMIN_ROLE();
+    admin = await impersonate(await authorizer.getRoleMember(DEFAULT_ADMIN_ROLE, 0), fp(100));
+  });
+
+  context('without permissions', () => {
+    itRevertsSettingFee(FeeType.Yield, fp(0.0157));
+
+    itRevertsSettingFee(FeeType.Swap, fp(0.0126));
+  });
```

Comment:
We don't need to remove this, but conceptually it is a bit overkill.

We know the contract does not allow setting these values without permissions, as it has been thoroughly unit-tested. What we want to do rather is perform a big integration check to make sure that the whole provider -> vault -> collector -> authorizer flow will work correctly and doesn't blow up (and also as a simple reminder of which permissions need to be granted).

Reply:
Thanks; I thought about it too, but I decided to include the tests for completion. I'll take it into account for the next fork test.